### PR TITLE
Added geoCoordinates extension

### PR DESCRIPTION
**Pull request by dajbelshaw**
I don’t really know what I’m doing here, but I’ve had a go! Tried to use this as a guide: http://schema.org/GeoCoordinates

### CODE REVIEW

File: extensions/geoCoordinatesExtension/schema.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "latitude": {
+      "type": "string",
+      "format": "number"
+    },
+    "latitude": {
+      "type": "string",
+      "format": "number"
+    },
+    "required": ["description"],
```
Comment (ottonomy) [inline extensions/geoCoordinatesExtension/schema.json]: `"required": ["description", "latitude", "longitude"]`

File: extensions/geoCoordinatesExtension/schema.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "More Info Link",
+  "description": "An extension allowing for the addition of the geographic location where an individual earned a badge.",
+  "type": "object",
+  "properties": {
+    "description": {
+      "type": "string",
+      "format": "text"
```
Comment (ottonomy) [inline extensions/geoCoordinatesExtension/schema.json]: "format" is a weird part of JSON-schema. See [this explanation](http://spacetelescope.github.io/understanding-json-schema/reference/string.html#format). It's not required to be supported by implementations.

Comment (dajbelshaw) [inline extensions/geoCoordinatesExtension/schema.json]: Cheers, have updated!

**Comment by kayaelle** [comment]
Thanks for contributing this Doug! In my thinking about a location extension I've been on the fence about there being separate extensions for geolocation, physical address and online location or just a single location extension.

The pros of keeping them separate - each one is simple
The pros of putting them together in one extension (with a minimum of one of the types required) is that there only needs to be on extension to think about.

The reason physical address would be a good extension in addition to geo location is that most users won't know what the coordinates will be when looking for an opportunity. The plus of geolocation is that you can build awesome maps. 

Also, online location - because some opportunities are only online.

Additionally, we may want to think about the extension or extensions having an array of locations ofr instances where opportunities are in more than one place.

What do you think?

Thanks

**Comment by threeqube** [comment]
In any case this is awesome @dajbelshaw! :+1:

**Comment by knightlabs** [comment]
@kayaelle re: 

> Additionally, we may want to think about the extension or extensions having an array of locations ofr instances where opportunities are in more than one place.

Is the use case there that if I earn something that _could_ have been earned in multiple locations that the instance would contain all of those locations, not just the place I earned it?

I'm trying to think of examples for why this would be useful or good, and not having success – any help on potential situations where this is ++?

OTOH, I can easily see a geolocation array as _evidence_ (the "I visited every major league baseball park" badge)

**Comment by ottonomy** [comment]
@knightlabs @kayaelle @dajbelshaw: I see multiple purposes for a location extension. 

* BadgeClass - Show location(s) where a badge might be available to earn.
* Assertion - Show location where a particular badge instance was earned.

I think we should get explicit about use cases for location in the documentation for a location extension, so that any issuer or consumer interested in it can be clear about what the intent of adding location to a badge object is. If we align on the above two use cases, I think the same extension could serve both these purposes. If there are other or additional reasons we try to target with location, we should be explicit about those and see if they can all fit together in one extension.

**Comment by threeqube** [comment]
I agree with the above two use cases and think we should start with those.

**Comment by threeqube** [comment]
Tangentially related: https://medium.com/@Esri/maps-we-love-be0a9d4f59b8

**Comment by dajbelshaw** [comment]
Great to see so much talent weigh in here. I'm a bit out of my depth, but will follow the conversation with interest to try and learn something to be useful in future! :)

**Comment by ottonomy** [comment]
The BA Standard Working Group cabinet has put this pull request approval on the agenda for next Tuesday's meeting (10:30am PDT (GMT-7)). @kayaelle is going to work on the code to get it ready to go before then. Thanks @dajbelshaw for putting it together!

**Comment by threeqube** [comment]
Agenda here fyi: http://etherpad.badgealliance.org/ba-standard-may26

**Comment by dajbelshaw** [comment]
This is great. Looking forward to this becoming a reality! :)

**Comment by kayaelle** [comment]
Hey all - to track back to my own question after talking it over some with @threeqube & @ottonomy we figured it would be better to have the separate extensions for geolocation & physical location. Also, for now, just one location instead of an array. If there becomes a need for an array of locations in the badge class, we could make adjustments then.

@dajbelshaw I'll be in touch soon to make some adjustments. Thank you!

**Comment by kayaelle** [comment]
Emailed @dajbelshaw with some adjustments and then we're good to go!

**Comment by ottonomy** [comment]
@kayaelle Could you give us an update on the status of this PR? I thought we were waiting on a couple final changes before pressing the big green button (or in this case, manually merging).

**Pull request closed without merging**